core/services/run_manager.go

```diff
@@ -21,20 +21,26 @@ import (
 )
 
 var (
-	numberRunsExecuted = promauto.NewCounter(prometheus.CounterOpts{
-		Name: "run_manager_runs_started",
-		Help: "The total number of runs that have run",
-	})
-	numberRunsResumed = promauto.NewCounter(prometheus.CounterOpts{
-		Name: "run_manager_runs_resumed",
-		Help: "The total number of run resumptions",
-	})
-	numberRunsCancelled = promauto.NewCounter(prometheus.CounterOpts{
-		Name: "run_manager_runs_cancelled",
-		Help: "The total number of run cancellations",
-	})
+	promTotalRunUpdates = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "run_manager_status_update_total",
+		Help: "The total number of status updates for Job Runs",
```

Comment:
s/Runs/Specs/ right? That's the grouping.

Reply:
That's odd, it should be job runs from what I see. The instances of the Job Specs executions (Job Runs) are what change statuses.

Reply:
Sure, but this metric would contain a total number of status updates for a job spec, since they are grouped by job spec - if this instead grouped by Job Run, as this help suggested, then you'll need to change it to Job Spec.

Reply:
Thing is though Job Specs do not get status updates.  Also, they aren't grouped, but labeled, they can be grouped by JobSpec or Status, but they are updates on Job Runs.

Reply:
I couldn't find documentation on it, but its similar to this -> https://github.com/prometheus/client_golang/blob/master/prometheus/timer.go#L30

Reply:
I don't see the need or the usage of an interface here?